iOS/ViewControllers/Courses/CourseViewController.swift

```diff
@@ -298,10 +318,37 @@ class CourseViewController: UIViewController {
         self.closeCourse()
     }
 
-    @IBAction private func shareCourse(_ sender: UIBarButtonItem) {
+    @IBAction private func showActionMenu(_ sender: UIBarButtonItem) {
+        var actions = [self.shareCourseAction]
+
+        if self.course.hasEnrollment {
+            actions.append(self.showCourseDatesAction)
+        }
+
+        let alert = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+        alert.popoverPresentationController?.barButtonItem = sender
+
+        for action in actions {
+            alert.addAction(action)
+        }
```

Comment:
No need to create a list of actions :)

```suggestion
        let alert = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
        alert.popoverPresentationController?.barButtonItem = sender
        
        alert.addAction(self.shareCourseAction)

        if self.course.hasEnrollment {
            alert.addAction(self.showCourseDatesAction)
        }
```